merkato/merkato.py

```diff
@@ -9,15 +9,55 @@
 
 
 class Merkato(object):
-    def __init__(self, configuration, ticker, spread):
+    def __init__(self, configuration, ticker, spread, ask_budget, bid_budget):
         self.exchange = Exchange(configuration)
         self.distribution_strategy = 1
         self.ticker = ticker # i.e. 'XMR_BTC'
         self.spread = spread # i.e '15
+        self.ask_profile = [(1, 5), (2, 5), (3, 10), (4, 15), (5, 20), (6, 25), (10, 30)] # (% change in price, % balance allocation)
+        self.bid_profile = [(-1, 5), (-2, 5), (-3, 10), (-4, 15), (-5, 20), (-6, 25), (-10, 30)]  # (% change in price, % balance allocation)
+        self.ask_budget = ask_budget
+        self.bid_budget = bid_budget
 
     def rebalance_orders(self):
         pass
 
+    def create_relative_bid_ladder(self,):
+        # a ladder that can be user configured (via gui)
+
+        # get market price to make ladder relative from
+        orders = self.exchange.get_all_orders(self.ticker)
+        lowest_ask = orders['asks'][0][0]
+
+        for level, allocation in self.bid_profile:
+            new_price = lowest_ask * (1 + (float(level) / 100.0))
+            new_amount = float(self.bid_budget) * (float(allocation) / 100.0)
+            if new_price < lowest_ask and new_amount < self.bid_budget:   # sanity check
```

Comment:
extract to a variable

Reply:
Sorry, what?

Reply:
The sanity check should be a variable declaring what the if check validates in english.

Reply:
Ok, as a general rule of thumb any evaluation statements that have more than one clause or a single lookup on a dictionary should be extracted as a declarative variable.